tests/at_tests.cpp

```diff
@@ -26,39 +26,39 @@ SUITE(at_tests)
 {
     TEST(static_array)
     {
-        int a[] = { 1, 2, 3, 4 };
+        size_t a[] = { 1, 2, 3, 4 };
 
-        for (int i = 0; i < 4; ++i)
+        for (size_t i = 0; i < 4; ++i)
             CHECK(at(a, i) == i+1);
```

Comment:
Yes, clang has the same warning (and some others I might add, but we can address those in a different PR)

```
/home/user/gsl/tests/at_tests.cpp:62:25: warning: implicit conversion changes signedness: 'int' to 'size_t' (aka 'unsigned long')
      [-Wsign-conversion]
            CHECK(at(a, i) == i+1);
```